lib/mapper/custom_field_mapper.rb

```diff
@@ -2,25 +2,28 @@ module BigcommerceProductAgent
     module Mapper
         class CustomFieldMapper
 
-            def self.map(field_map, product, bc_product)
+            def self.map(field_map, raw_product, bc_product, current_custom_fields, namespace)
                 fields = {
                     upsert: [],
                     delete: [],
                 }
 
                 existing_fields = {}
 
-                if bc_product
-                    bc_product['custom_fields'].each do |cf|
+                if bc_product && current_custom_fields
+                    current_custom_fields.each do |cf|
                         cf['product_id'] = bc_product['id']
                         existing_fields[cf['name'].to_s] = cf
                     end
                 end
 
                 if field_map && field_map['additionalProperty']
```

Comment:
This is only mapping custom fields for `additionalProperty` values.

This will miss things like the `disambiguatingDescription => subtitle` mapping defined in the agent options. Any key in `custom_fields_map` that is _not_ nested as an `additionalProperty` should also be mapped as its own custom field

Reply:
I believe the other fields are being taken care of starting at line 30

Reply:
Ah -- you are correct. Not sure what I was looking at.